drm/drm_fbdev_generic.h: No such file or directory

Hello i am using  debian 12, i followed the instructions as per the read me file. and i am getting the error below
![image](https://github.com/rhgndf/fl2000_drm/assets/47003569/3092690a-8a18-4481-9e1d-02e2394c8fd4)

same here:
```
ragnarok@ragnarok:~/src/fl2000_drm-rhgndf$ make
make CHECK="/usr/bin/sparse" -C /lib/modules/6.1.0-20-amd64/build M=/home/ragnarok/src/fl2000_drm-rhgndf modules
make[1]: se entra en el directorio '/usr/src/linux-headers-6.1.0-20-amd64'
  CC [M]  /home/ragnarok/src/fl2000_drm-rhgndf/fl2000_drv.o
  CC [M]  /home/ragnarok/src/fl2000_drm-rhgndf/fl2000_registers.o
  CC [M]  /home/ragnarok/src/fl2000_drm-rhgndf/fl2000_interrupt.o
  CC [M]  /home/ragnarok/src/fl2000_drm-rhgndf/fl2000_streaming.o
  CC [M]  /home/ragnarok/src/fl2000_drm-rhgndf/fl2000_connector.o
  CC [M]  /home/ragnarok/src/fl2000_drm-rhgndf/fl2000_i2c.o
  CC [M]  /home/ragnarok/src/fl2000_drm-rhgndf/fl2000_drm.o
/home/ragnarok/src/fl2000_drm-rhgndf/fl2000_drm.c:15:10: fatal error: drm/drm_fbdev_generic.h: No existe el fichero o el directorio
   15 | #include <drm/drm_fbdev_generic.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [/usr/src/linux-headers-6.1.0-20-common/scripts/Makefile.build:255: /home/ragnarok/src/fl2000_drm-rhgndf/fl2000_drm.o] Error 1
make[1]: *** [/usr/src/linux-headers-6.1.0-20-common/Makefile:2035: /home/ragnarok/src/fl2000_drm-rhgndf] Error 2
make[1]: se sale del directorio '/usr/src/linux-headers-6.1.0-20-amd64'
make: *** [Makefile:18: modules] Error 2
ragnarok@ragnarok:~/src/fl2000_drm-rhgndf$ uname -a
Linux ragnarok 6.1.0-20-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.85-1 (2024-04-11) x86_64 GNU/Linux

 ```

--- a/fl2000_connector.c
+++ b/fl2000_connector.c
@@ -27,12 +27,12 @@ static int fl2000_read_edid(void *data, u8 *buf, unsigned int block, size_t len)
 }
 static int fl2000_get_modes(struct drm_connector *connector)
 {
-	struct edid *edid;
+	const struct drm_edid *edid;
 	int ret;
 
-	edid = drm_do_get_edid(connector, fl2000_read_edid, connector->ddc);
-	drm_connector_update_edid_property(connector, edid);
-	ret = drm_add_edid_modes(connector, edid);
+	edid = drm_edid_read_custom(connector, fl2000_read_edid, connector->ddc);
+	drm_edid_connector_update(connector, edid);
+	ret = drm_edid_connector_add_modes(connector);
 	kfree(edid);
 
 	//ret = drm_add_modes_noedid(connector, 1920, 1200);

--- a/fl2000_drm.c
+++ b/fl2000_drm.c
@@ -4,6 +4,7 @@
  * (C) Copyright 2018-2020, Artem Mygaiev
  */
 
+#include <linux/version.h>
 #include <linux/dma-buf.h>
 #include <linux/printk.h>
 #include <linux/usb.h>
@@ -12,7 +13,11 @@
 #include <drm/drm_damage_helper.h>
 #include <drm/drm_drv.h>
 #include <drm/drm_fb_helper.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,11,0)
+#include <drm/drm_fbdev_ttm.h>
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0)
 #include <drm/drm_fbdev_generic.h>
+#endif
 #include <drm/drm_fourcc.h>
 #include <drm/drm_framebuffer.h>
 #include <drm/drm_gem_atomic_helper.h>
@@ -566,7 +571,11 @@ int fl2000_drm_init(struct fl2000 *fl2000_dev)
 	fl2000_reset(usb_dev);
 	fl2000_usb_magic(usb_dev);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,11,0)
+	drm_fbdev_ttm_setup(drm, FL2000_FB_BPP);
+#else
 	drm_fbdev_generic_setup(drm, FL2000_FB_BPP);
+#endif
 
 	return 0;
 

--- a/fl2000_streaming.c
+++ b/fl2000_streaming.c
@@ -12,6 +12,7 @@
  */
 
 #include <linux/scatterlist.h>
+#include <linux/vmalloc.h>
 
 #include <drm/drm_managed.h>
 #include <drm/drm_vblank.h>

